contracts/truefi2/DeficiencyToken.sol

```diff
@@ -30,8 +30,8 @@ contract DeficiencyToken is IDeficiencyToken, ERC20 {
         _mint(address(_loan.pool()), _amount);
     }
 
-    function burn(uint256 amount) external override {
-        _burn(msg.sender, amount);
+    function burnFrom(address account, uint256 amount) external override {
+        _burn(account, amount);
```

Comment:
This lacks one line:
`_approve(account, _msgSender(), allowance(account, _msgSender()).sub(amount))`

Without it, anyone can burn anyones tokens without any approvals. In my opinion it is also worth writing some tests for this function.